libsoftwarecontainer/src/gateway/gateway.cpp

```diff
@@ -21,11 +21,30 @@
 
 namespace softwarecontainer {
 
+
+Gateway::Gateway(const std::string &id,
+                 std::shared_ptr<ContainerAbstractInterface> container,
+                 bool quickLaunchEnabled) :
+    m_id(id),
+    m_container(container),
+    m_quickLaunchEnabled(quickLaunchEnabled),
+    m_state(GatewayState::CREATED)
+{
+}
+
+std::string Gateway::id() const
+{
+    return m_id;
+}
+
 bool Gateway::setConfig(const json_t *config)
 {
-    if (m_state == GatewayState::ACTIVATED) {
-        log_error() << "Can not configure a gateway that is already activated: " << id();
-        throw GatewayError("Gateway already activated");
+    if (m_state == GatewayState::ACTIVATED && !m_quickLaunchEnabled) {
```

Comment:
To reduce the risk of an accidental assignment? In principle I agree, it's just that the whole code base (basically) does it like this. I didn't plan to go through the gateway code in hunt for this type of improvement. Do you think we could perhaps have that as a separate change?